src/datasets/utils/download_manager.py

```diff
@@ -122,9 +122,7 @@ def upload(local_file_path):
 
     def _record_sizes_checksums(self, url_or_urls, downloaded_path_or_paths):
```

Comment:
Can we start adding type hints here ? Especially since the variable names are the same as in `.download` but the type is different: here they're NestedDataStructure objects

Reply:
Sure!